fdbserver/ResolutionBalancer.actor.cpp

```diff
@@ -0,0 +1,187 @@
+/*
+ * ResolutionBalancer.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbserver/ResolutionBalancer.actor.h"
+
+#include "fdbclient/KeyRangeMap.h"
+#include "fdbserver/MasterInterface.h"
+#include "fdbserver/Knobs.h"
+#include "flow/flow.h"
+
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+void ResolutionBalancer::setResolvers(const std::vector<ResolverInterface>& v) {
+	resolvers = v;
+	if (resolvers.size() > 1)
+		triggerResolution.trigger();
```

Comment:
It would be a good idea to maintain stats regarding how often "resolution is getting triggered". 

Further, was thinking how we can automate the testing on this module, one possible way would be subject a workload that is expected to trigger "rebalancing" and at the end of test assert that "rebalancing-counter" value > 1 ( or expected count). 

It is kinda scary that with last refactor the module stopped functioning and no test were able to detect it (thanks to you for noticing it). 
Thoughts? (Again, we can take it as a separate PR)

Reply:
Agree a workload is probably the best option. The "rebalancing-counter" is probably something in `ResolutionBalancer::setChangesInReply`, where I added a `TEST` macro, line 42.